WooCommerce/Classes/POS/Presentation/TotalsView.swift

```diff
@@ -89,39 +90,47 @@ private extension TotalsView {
     var totalsFieldsView: some View {
         HStack(alignment: .center) {
             Spacer()
-            VStack() {
-                subtotalFieldView(title: Localization.subtotal,
-                                  formattedPrice: viewModel.formattedCartTotalPrice,
-                                  shimmeringActive: viewModel.isShimmering,
-                                  redacted: viewModel.isSubtotalFieldRedacted,
-                                  matchedGeometryId: Constants.matchedGeometrySubtotalId)
-                Spacer().frame(height: Constants.subtotalsVerticalSpacing)
-                subtotalFieldView(title: Localization.taxes,
-                                  formattedPrice: viewModel.formattedOrderTotalTaxPrice,
-                                  shimmeringActive: viewModel.isShimmering,
-                                  redacted: viewModel.isTaxFieldRedacted,
-                                  matchedGeometryId: Constants.matchedGeometryTaxId)
-                Spacer().frame(height: Constants.totalVerticalSpacing)
-                Divider()
-                    .overlay(Constants.separatorColor)
-                Spacer().frame(height: Constants.totalVerticalSpacing)
-                totalFieldView(formattedPrice: viewModel.formattedOrderTotalPrice,
-                               shimmeringActive: viewModel.isShimmering,
-                               redacted: viewModel.isTotalPriceFieldRedacted,
-                               matchedGeometryId: Constants.matchedGeometryTotalId)
+            switch posModel.orderState {
+            case .idle,
+                    .syncing,
+                    .error:
+                totalsFields(orderTotals: nil)
+            case .loaded(let orderTotals):
+                totalsFields(orderTotals: orderTotals)
             }
-            .padding(Constants.totalsLineViewPadding)
-            .frame(minWidth: Constants.pricesIdealWidth)
-            .fixedSize(horizontal: true, vertical: false)
             Spacer()
         }
     }
 
+    @ViewBuilder func totalsFields(orderTotals: PointOfSaleOrderTotals?) -> some View {
+        let totalsLoading = orderTotals == nil
+        VStack {
+            subtotalFieldView(title: Localization.subtotal,
+                              formattedPrice: orderTotals?.cartTotal,
+                              shimmeringActive: totalsLoading,
```

Comment:
I've replaced the two parameters (redacted is the other) with one, because we only ever used them together to show the loading state